lib/scheduled_publishing_checker.rb

```diff
@@ -0,0 +1,76 @@
+class ScheduledPublishingChecker
+  MISSING_EDITION = 'missing_edition'.freeze
+  MISSING_REDIRECT = 'missing_redirect'.freeze
+  SUCCESSFULLY_PUBLISHED = 'successfully_published'.freeze
+
+  # scheduled tasks can only run at 0, 15, 30 and 45 minutes past the hour.
+  # This job should be run scheduled to run 30seconds later so that documents
+  # have a chance to be published first. In addition this job should run every
+  # minutes to check if the document has now appeared - this is inline with the
+  # healthcheck frequency
+  def check
+    start_date = MissedScheduledPublishing.maximum(:scheduled_publication) || 1.week.ago
+
+    since_last_run = Edition.includes(:document).where("scheduled_publication > ? and scheduled_publication < ?", start_date, Time.zone.now)
+
+    since_last_run.each do |edition|
+      url = Whitehall.url_maker.public_document_url(edition)
+      announcement_url = edition.respond_to?(:statistics_announcement) && edition.statistics_announcement.public_path
+
+      status = check_url(url, announcement_url)
+
+      MissedScheduledPublishing.create(
+        url: url,
+        announcement_url: announcement_url,
+        scheduled_publication: edition.scheduled_publication,
+        status: status,
+        found_at_first_attempt: status == SUCCESSFULLY_PUBLISHED,
+      )
+    end
```

Comment:
Feels weird to me to create all the instances and check them, only to immediately after this loop also check all the ones that haven't succeeded.  Could we create them in a "just added" state in this loop and then pick them up with the rest of them in the next loop.

Or is the intention that new things do get two checks in quick succession?